### DISCUSSION TITLE
Add Pony Preset?

### DISCUSSION THREAD

**Discussion by jaimeastorga2000** [lllyasviel/Fooocus#3182] [Ideas]
[Pony](https://civitai.com/models/257749/pony-diffusion-v6-xl) and its [derivatives](https://civitai.com/models/288584/autismmix-sdxl) have _completely_ revolutionized non-realistic SDXL generation. With a little practice and the right LoRAs, they can make anything from pixel art to cartoons to comics to manga to anime (_actual_ anime, not that 2.5-D stuff that models like [Blue Pencil](https://civitai.com/models/119012/bluepencil-xl) and [Anima Pencil](https://civitai.com/models/261336/animapencil-xl) generate), showing particular aptitude for NSFW prompt comprehension. Should Fooocus add a Pony preset? There could be a Pony style with all the standard tags ([score_9](https://civitai.com/articles/4248/what-is-score9-and-how-to-use-it-in-pony-diffusion), source_anime, etc.) checked by default to make it easy to use.

**Comment by ILikeToasters** [comment]
I guess my first question would be, besides a style what else would be preferred for Pony? A style is not a bad idea. I am curious since I have experimented a tiny bit when people have asked about it but not really much further than that.

**Reply by AnalyticsXXI** [reply]
- CFG can be increased a bit
- score_2, score_3, score_4 can be putted in negative prompt by default, since they are always supposed to be there for Pony
- maybe changing sampler to Euler, not sure if it's better for Pony or not
- if you will ever want to change default CLIP to 1, you will be able to keep it 2 for Pony, as necessary.

**Reply by BoJangles002** [reply]
>I guess my first question would be, besides a style what else would be preferred for Pony?

Pony based models have an issue with creating random artifacts and the setting used to fix it in other UIs doesn't seem to exist in Fooocus. In A1111 you "enable emphasis." In Forge you turn emphasis to "no norm." I can't find any similar setting in Fooocus.

**Comment by PeterJared** [comment]
I suppose you mean affecting these values. But where is the "score"?  Also, when you say, CFG can be increased, over what number?
![image](https://github.com/lllyasviel/Fooocus/assets/49351356/54b18feb-0e8c-49b9-b894-8a1c6e387fb8)
Can you send a screenshot highlighting the final numbers you changed in the UI

**Reply by AnalyticsXXI** [reply]
No, CFG in Foocus called "guidance scale" (on the very top of the page), and score_4 is just a keyword for the negative prompt. It's a pony thing, means "overall quality of image 4/10". So it's the same as putting "worst quality" in the negative prompt.

**Comment by BoJangles002** [comment]
In regards to pony support, pony has two glaring issues on Fooocus. It's possible they're the same root issue. Appreciate any response about possibility of a fix or if someone with better understanding can open up an issue on it. 


First problem is shown in this image. 
![artifacts](https://github.com/lllyasviel/Fooocus/assets/48421024/cea872c3-a09c-4933-824d-e6a112fd0cae)
If you hit certain weighting of prompts/tokens, it turns the image into total noise. In a1111 you enable "emphasis." In forge you turn emphasis to "no norm."
The issue is discussed in detail in [this article](https://civitai.com/articles/4044). 
There is also [a link](https://github.com/hako-mikan/sd-webui-prevent-artifact) to the first extension to fix the issue.

Secondly pony based models are unable to use inpaint engine. The image below was inpainted with the same exact settings. No errors are thrown. 
The left is the result with a pony based model, the right is the default model animapencil. 

![ponyvsanima](https://github.com/lllyasviel/Fooocus/assets/48421024/a3c3c876-56f9-456a-971c-e53249245079)

The images look VERY similar at early steps, but at roughly 50% step count, pony based models stagnate and appear to endlessly shuffle noise.

**Reply by ILikeToasters** [reply]
For the second part you can set the Fooocus inpaint engine to none in the debug menu.

**Reply by BoJangles002** [reply]
Correct but then you lose all the context that inpaint engine brings. It's impossible to outpaint coherently without inpaint engine for instance. My hope is a fix can be brought in for the first issue and it will resolve the inpaint engine issue.

**Reply by vvytch** [reply]
The inpaint problem only occured to me after a certain unnoticable update that changed the **Forced Overwrite of Refiner Switch Step** to **-1** by default. If you change it to **1** instead, inpaint seems to work with pony models!

**Reply by vitorfdl** [reply]
confirmed that changing Overwrite Refiner Switch to 1 makes pony inpainting work.

default_overwrite_switch for setting in the preset
